Sources/Sentry/SentryRateLimitCategoryMapper.m

```diff
@@ -0,0 +1,18 @@
+#import <Foundation/Foundation.h>
+#import "SentryRateLimitCategoryMapper.h"
+#import "SentryRateLimitCategory.h"
+#import "SentryEnvelopeItemType.h"
+
+@interface SentryRateLimitCategoryMapper ()
+
+@end
+
+@implementation SentryRateLimitCategoryMapper
+
++ (NSString *_Nonnull)mapEventTypeToCategory:(NSString *)eventType {
+    // Currently we classify every event type as error.
+    // This is going to change in the future.
+    return SentryRateLimitCategoryError;
+}
```

Comment:
`high:` I'd like to double-check this with @jan-auer 
I believe we can fallback everything to `error`, but we should map and use the categories that we already know.
eg `session` as it already matches the envelope item type

Reply:
The plan is to use this function only for events but not envelopes. Releasing this in GA as it is would be a very bad idea I agree, but the current implementation isn't any better, because it uses `event`. So this PR improves it a little bit and I am going to work on this problem today. Are you fine with merging this and I improve it in another PR?

Reply:
yes, I am fine with it if a new PR would come.
let's just keep in mind that we can't deploy like this, or sessions would be rate-limited.

one thing to note is: it's hard to review the whole picture of rate-limiting because it's broken in several PRs, so I don't get how they would connect altogether.

Small PRs are definitely a good thing and easier to review, but reading this PR makes me think if we are not missing something as we can't really read the whole code, we can discuss this offline.

I'll mark it as resolved.

Reply:
I agree this is one downside of smaller PRs but we can address this with a different strategy rather than making big single PRs.

We can do a main PR to hold the complete feature and open small PRs that address parts of it that can be tested and written independently to merge not to `master` but to this intermediate PR`. 

When time comes to merge the complete PR, we can do a merge commit not to lose the history. So a big feature with 1k lines of code would be perhaps 5 PRs straight into the repo history. I'd say that's a better approach than opening a single PR with 1k lines directly, also because takes time to write such PR and lots of code could get early feedback.